pkg/controllers/report/aggregate/controller.go

```diff
@@ -371,8 +374,11 @@ func (c *controller) frontReconcile(ctx context.Context, logger logr.Logger, _,
 		return nil
 	}
 	// try to find the owner
-	if c.adopt(ctx, reportMeta) {
+	if adopted, forbidden := c.adopt(ctx, reportMeta); adopted {
 		return nil
+	} else if forbidden {
+		logger.Info("deleting because insufficient permission to fetch resource")
```

Comment:
Shouldn't we fail the policy upon its creation?

Reply:
looks like it sends warnings: https://github.com/kyverno/kyverno/blob/b666b1665ce421aca008bb1b94563ccde41488bd/pkg/policy/validate/validate.go#L144-L148

Reply:
Can we return warnings for mutate rules as well?

Reply:
I will have to pass the name of reports controller SA to admission controller so the logic will work with helm install

Reply:
It's already there:
```
            - --reportsServiceAccountName=system:serviceaccount:{{ include "kyverno.namespace" . }}:{{ include "kyverno.reports-controller.serviceAccountName" . }}
```

Reply:
@realshuting added
```
$ kubectl apply -f chainsaw-step-01-apply-1-1.yaml                                                               2s 
Warning: system:serviceaccount:kyverno:kyverno-reports-controller requires permissions get,list,watch for resource Secret
```